doc/developer/reference/README.md

```diff
@@ -0,0 +1,21 @@
+# Materialize Platform Reference Documentation
+
+This space is meant to be an authoritative source for the following types of information:
+
+-   A high-level architecture overview of the various bits of Materialize Cloud.
```

Comment:
Is this documentation intended also for the cloud part or only the database part of Materialize? I assumed the latter, which is why "Materialize *Cloud*" confuses me.

Reply:
I used the wrong term by mistake, good catch.